### PR TITLE
Backend changes for whites space character bug in target space.

### PR DESCRIPTION
**Pull request by KristianShishoev**

### CODE REVIEW

File: com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/cf/v1_0/CloudModelBuilderTest.java [hunk 1]
```diff
@@ -332,8 +332,7 @@ public void setUp() throws Exception {
         DescriptorHandler handler = getDescriptorHandler();
 
         String targetName = extensionDescriptor.getDeployTargets().get(0);
-        Target implicitTarget = getDeployTargetFactory().create(targetName, platforms.get(0).getName());
-        Target target = handler.findTarget(targets, targetName, implicitTarget);
+        Target target = handler.findTarget(targets, targetName, null);
```
Comment (monochronique) [inline com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/cf/v1_0/CloudModelBuilderTest.java]: Why is this changed to null? The DetectTargetStep still computes an implicit target during its execution.

File: com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/validators/MtaArchiveValidatorTest.java [hunk 2]
```diff
@@ -35,7 +36,7 @@
 
     private static final long MAX_MTA_DESCRIPTOR_SIZE = 1024 * 1024l; // 1 MB
 
-    private static final String XS2_PLATFORM_NAME = "initial initial";
+    private static final String XS2_PLATFORM_NAME = "XS2-INITIAL";
```
Comment (monochronique) [inline com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/validators/MtaArchiveValidatorTest.java]: Why is this changed?

File: com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/PublishConfigurationEntriesStepTest.java [hunk 3]
```diff
@@ -81,7 +82,8 @@ public void setUp() throws Exception {
     public void prepareDao() throws Exception {
         for (ConfigurationEntry entry : exisitingConfigurationEntries) {
             Mockito.when(configurationEntryDaoMock.find(Mockito.matches(entry.getProviderNid()), Mockito.matches(entry.getProviderId()),
-                Mockito.matches(entry.getProviderVersion().toString()), Mockito.matches(entry.getTargetSpace()), Mockito.any(),
+                Mockito.matches(entry.getProviderVersion().toString()),
+                Mockito.any(CloudTarget.class), Mockito.any(),
```
Comment (monochronique) [inline com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/PublishConfigurationEntriesStepTest.java]: In this way the test no longer validates correct filtration based on the target. Can this be enhanced to take into consideration the ConfigurationEntry's target?

File: com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/resources/ConfigurationEntriesResource.java [hunk 4]
```diff
@@ -165,9 +165,14 @@ public Response getConfigurationEntry(@PathParam(ID) long id) throws SLException
     @POST
     @Consumes(MediaType.APPLICATION_XML)
     public Response createConfigurationEntry(String xml) throws SLException {
+        
         ConfigurationEntryDto dto = parseDto(xml, CREATE_CONFIGURATION_ENTRY_SCHEMA_LOCATION);
-
         ConfigurationEntry result = entryDao.add(dto.toConfigurationEntry());
+
```
Comment (monochronique) [inline com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/resources/ConfigurationEntriesResource.java]: The exception here is thrown after inserting the result in the DAO. Could this potentially lead to a corrupted database content?

File: com.sap.cloud.lm.sl.cf.web/src/test/java/com/sap/cloud/lm/sl/cf/web/resources/ConfigurationEntriesResourceTest.java [hunk 5]
```diff
@@ -53,52 +54,65 @@ public ConfigurationEntriesResourceTest(TestCase<TestInput> test) {
...
                 new SearchRequestTest(new SearchRequestTestInput(Arrays.asList("{\"foo\":\"bar\",\"baz\":\"qux\"}"), "R:parsed-properties-01.json"), "R:configuration-entries-resource-test-output-08.json"),
             },
-            // (12)
+            // (13)
             {
                 new SearchRequestTest(new SearchRequestTestInput(Arrays.asList("a"), "R:parsed-properties-01.json"), "E:Could not parse content query parameter as JSON or list"),
             },
+            // (14)
+            {
+                new PostRequestTest(new PostRequestTestInput("configuration-entry-06.xml"), "R:configuration-entries-resource-test-output-01.json"),
```
Comment (monochronique) [inline com.sap.cloud.lm.sl.cf.web/src/test/java/com/sap/cloud/lm/sl/cf/web/resources/ConfigurationEntriesResourceTest.java]: Should there be a test case where the user specifies both "target-space" and "provider-target"? What would happen then? This seems to be a valid input scenario according to the xsd schema.

Comment (KristianShishoev) [reply com.sap.cloud.lm.sl.cf.web/src/test/java/com/sap/cloud/lm/sl/cf/web/resources/ConfigurationEntriesResourceTest.java]: It must take take parameters from provider-target section but i added test just in case.

File: com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/validators/MtaArchiveValidatorTest.java (hunk 2 above)
Comment (KristianShishoev) [reply com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/validators/MtaArchiveValidatorTest.java]: Code must cover both cases but just in case i added tests for both scenarios.

File: com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/cf/v1_0/CloudModelBuilderTest.java (hunk 1 above)
Comment (KristianShishoev) [reply com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/cf/v1_0/CloudModelBuilderTest.java]: Because all targets are described in targets.json and in tests all targets are referenced from this file.
So this implicit target will never work because this create method requires org and space parameters.

File: com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/liquibase/TransformFilterColumn.java [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+    Map<Long, String> transformData(Map<Long, String> retrievedData) {
+
+        Map<Long, String> transformedData = new HashMap<Long, String>();
+        for (Map.Entry<Long, String> entry : retrievedData.entrySet()) {
+
+            if (StringUtils.isEmpty(entry.getValue())) {
+                continue;
+            }
+            JsonElement initialFilterJsonElement = new JsonParser().parse(entry.getValue());
```
[the remark this answers is not in the document]
Comment (KristianShishoev) [reply com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/liquibase/TransformFilterColumn.java]: No need. The code is clear in this way also.
